python/graphstorm/model/edge_decoder.py

```diff
@@ -909,7 +911,58 @@ def predict_proba(self, g, h, e_h):
         return out
 
 ##################### Link Prediction Decoders #######################
-class LinkPredictNoParamDecoder(GSLayerNoParam):
+class LinkPredictionTestScoreInterface(abc.ABC):
```

Comment:
Shall we add this class in the API doc?

Reply:
An end user would need to implement either `LinkPredictNoParamDecoder` or a `LinkPredictLearnableDecoder`, so we can just keep those two in the API. There's not much value in this interface in isolation, it needs to be part of a `*Decoder`